resources/js/components/Listing/Listing.vue

```diff
@@ -43,6 +46,22 @@
                 filters: this.filters,
                 sortOptions: this.sortOptions,
                 reactiveFilters: this.reactiveFilters,
+                variables: this.variables
+            })
+        },
+
+        methods: {
+            initPageSize() {
+                this.variables.pageSize = Turbo.navigator.location.searchParams?.get('pageSize') ?? this.$root.config.grid_per_page
+            }
+        },
+
+        created() {
+            this.initPageSize()
+            this.$watch('variables.pageSize', (pageSize) => {
```

Comment:
Watchers can be defined within `watch: {}`

Reply:
I've now put the watcher inside `watch()`. But because `pageSize` is only reactive within `variables` and `watch: {}` can't listen to `variables.pageSize`, I have now created a computed property for pageSize.